instrumentation/runtime-telemetry-jfr/library/src/main/java/io/opentelemetry/instrumentation/runtimetelemetryjfr/internal/RecordedEventHandler.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimetelemetryjfr.internal;
+
+import io.opentelemetry.instrumentation.runtimetelemetryjfr.JfrFeature;
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import jdk.jfr.consumer.RecordedEvent;
+
+/**
+ * Convenience/Tag interface for defining how JFR events should turn into metrics.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public interface RecordedEventHandler
+    extends Consumer<RecordedEvent>, Predicate<RecordedEvent>, Closeable {
```

Comment:
Consider adding a non-throwing `close()` declaration, like the SDK often does, e.g.: https://github.com/open-telemetry/opentelemetry-java/blob/ba06aa777cb371b5bebd9682fc4f8042c9cba88a/api/all/src/main/java/io/opentelemetry/api/metrics/ObservableLongCounter.java#L27-L28
Should make some error handling simpler

Reply:
Good idea! Applied.